core/src/com/unciv/logic/map/mapgenerator/MapRegions.kt

```diff
@@ -787,35 +808,41 @@ class MapRegions (val ruleset: Ruleset){
         // If there are any weightings defined in json, assume they are complete. If there are none, use flat weightings instead
         val fallbackWeightings = ruleset.tileResources.values.none {
             it.resourceType == ResourceType.Luxury &&
-                (it.hasUnique(UniqueType.LuxuryWeighting) || it.hasUnique(UniqueType.LuxuryWeightingForCityStates)) }
+                (it.uniqueObjects.any { unique -> unique.isOfType(UniqueType.ResourceWeighting) } || it.hasUnique(UniqueType.LuxuryWeightingForCityStates)) }
 
-        val maxRegionsWithLuxury = if (regions.count() > 12) 3 else 2
+        val maxRegionsWithLuxury = when {
```

Comment:
Regions.count/4... ?

Reply:
More like `max(1, min(3, (Regions.count() - 1) / 4))` but surely this is clearer?